### PR TITLE
[#9536] Instructor Course Edit Page E2E Test 

### PR DESCRIPTION
**Pull request by jtankw3**
Part of #9536

**Outline of Solution**

- Add  markCheckBoxAsUnchecked to AppPage 
- Add InstructorCourseEditPage to represent instructor course edit page
- Add InstructorCourseEnrollPageE2ETest which tests:

    - verify cannot edit without privilege
    - verify loaded data
    - add instructor
    - resend invite
    - edit instructor
    - delete instructor
    - edit course
    - delete course
- Add retry manager to verifyStatusMessage as it is error prone due to timing problems

### CODE REVIEW

File: src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        InstructorAttributes newInstructor = InstructorAttributes
+                .builder(course.getId(), "InsCrsEdit.test@gmail.tmt")
+                .withName("Teammates Test")
+                .withIsDisplayedToStudents(true)
+                .withDisplayedName("Instructor")
+                .withRole("Tutor")
+                .build();
+
+        editPage.addInstructor(newInstructor);
+        editPage.verifyStatusMessage("\"The instructor Teammates Test has been added successfully. "
```
Comment (rrtheonlyone) [inline src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java:74]: Why is this status message hardcoded? I think it should be programatically derived. 

If not, the test is brittle and any small change in the message (an extra space etc.) will break the E2E tests.

The same applies to the other status messages.

File: src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+        editPage.verifyCourseDetails(course);
+
+        editPage.verifyInstructorDetails(1, instructors[0]);
+        editPage.verifyInstructorDetails(2, instructors[1]);
+        editPage.verifyInstructorDetails(3, instructors[2]);
+        editPage.verifyInstructorDetails(4, instructors[3]);
+        editPage.verifyInstructorDetails(5, instructors[4]);
+
+        ______TS("add instructor");
```
Comment (rrtheonlyone) [inline src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java:64]: In this test case, we add a new instructor and verify that the frontend has the instructor's details. However, the backend is not verified right? We should also check that the database has the new instructor in this course.

File: src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        editPage.verifyStatusMessage("\"The instructor Teammates Test has been added successfully. "
+                + "An email containing how to 'join' this course will be sent to InsCrsEdit.test@gmail.tmt"
+                + " in a few minutes.\"");
+        editPage.verifyInstructorDetails(6, newInstructor);
+
+        ______TS("resend invite");
+        editPage.resendInstructorInvite(6);
+        editPage.verifyStatusMessage("An email has been sent to InsCrsEdit.test@gmail.tmt");
+
+        ______TS("edit instructor");
```
Comment (rrtheonlyone) [inline src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java:83]: Same for this test case. The backend is not verified. Actually this point holds throughout for the other test cases!

File: src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java (hunk 1 above)
Comment (wkurniawan07) [reply src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java:74]: This is actually a grey area. On one hand, what you said is of course true. However, on the other hand, this is a black-box test, meaning that the test code should not actually know that there exists some constant in the code that holds that string.

Either way is acceptable.

Comment (jtankw3) [reply src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java]: I did not use the Const values as they are Deprecated. I have added tried to use variables instead for names, emails etc in the status message instead to try to make is slightly less brittle.

File: src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java (hunk 2 above)
Comment (jtankw3) [reply src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java:66]: Added some methods with Backdoor to do some verification for backend

**Comment by wkurniawan07** [comment]
@jtankw3 it looks like you're (accidentally?) including other commits in your branch

**Comment by jtankw3** [comment]
> @jtankw3 it looks like you're (accidentally?) including other commits in your branch

I removed the extra wait also because I realized it is not needed and causing timeout

**Comment by jtankw3** [comment]
> @jtankw3 it looks like you're (accidentally?) including other commits in your branch

Not sure how that happened but I think they are gone now

**Pull request merged**